src/wp-includes/media.php

```diff
@@ -2029,6 +2029,22 @@ function wp_sizes_attribute_includes_valid_auto( string $sizes_attr ): bool {
 	return 'auto' === strtolower( trim( $first_size, " \t\f\r\n" ) );
 }
 
+/**
+ * Prints a CSS rule to fix potential visual issues with images using `sizes=auto`.
+ *
+ * This rule overrides the similar rule in the default user agent stylesheet, to avoid images that use e.g.
+ * `width: auto` or `width: fit-content` to appear smaller.
+ *
+ * @since 6.7.1
+ * @see https://html.spec.whatwg.org/multipage/rendering.html#img-contain-size
+ * @see https://core.trac.wordpress.org/ticket/62413
+ */
+function wp_print_auto_sizes_contain_css_fix() {
```

Comment:
This looks right. What would you think about bailing early if auto-sizes has been disabled via a filter similar to what is being proposed in https://github.com/WordPress/wordpress-develop/pull/7812?

Reply:
Makes sense to me. Let's either add this here or in #7812, depending on which of the two gets committed first. We can coordinate that once we're confident about the changes.

Reply:
Added in https://github.com/WordPress/wordpress-develop/pull/7813/commits/3b2a48efa513ebbc43853e0f3d9f32e9c743ecc3